src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java

```diff
@@ -16,14 +16,21 @@
 interface ConditionMapper {
 
     /**
-     * Maps expressions in the given request for a virtual table to a request for the physical table,
-     * based on the virtual table's primary key.
+     * Maps the update expression, condition expression, and expression attributes in the given virtual update request
+     * to the corresponding physical update request.
      */
-    void apply(RequestWrapper request);
+    void applyForUpdate(RequestWrapper request);
 
     /**
-     * Maps expressions in the given request for a virtual table to a request for the physical table,
+     * Maps the key condition in the given virtual query request to the corresponding physical key condition,
      * based on the given virtual secondary index if it's provided, or if not, on the virtual table primary key.
      */
-    void apply(RequestWrapper request, @Nullable DynamoSecondaryIndex virtualSecondaryIndex);
+    void applyToKeyCondition(RequestWrapper request, @Nullable DynamoSecondaryIndex virtualSecondaryIndex);
+
+    /**
+     * Maps the filter expression in the given virtual request to the corresponding physical filter expression,
+     * applying the change to either the primary or secondary expression in the request wrapper.
+     */
+    void applyToFilterExpression(RequestWrapper request, boolean isPrimaryExpression);
```

Comment:
Used to map filter/condition expressions in a put/delete/scan/query request, which in terms of RequestWrapper is primary in the case of put/delete/scan, and secondary in a query request.

Reply:
Seems like it may be useful to revisit the `RequestWrapper` abstraction in the context of expression mapping to make this interface more explicit (in a future change).

Reply:
Yeah... I was considering that too but ended up not doing it and limiting the amount of changes in this PR